docs/building-a-bento-amp-extension.md

```diff
@@ -480,7 +521,143 @@ Please note that for components that support all size-defined layouts, they also
 AmpBaseCarousel['layoutSizeDefined'] = true;
 ```
 
-## Experiments
+### Register element with AMP
+
+Once you have implemented your AMP element, you need to register it with
+AMP; all AMP component extensions include the `amp-` prefix.
+
+One way to specify the appropriate styles is to tell AMP which class to use for this tag name and which CSS to load.
+
+```javascript
+import {CSS} from '../../../build/amp-my-element-1.0.css';
+
+AMP.extension('amp-my-element', '1.0', (AMP) => {
+  AMP.registerElement('amp-my-element', AmpMyElement, CSS);
+});
+```
+
+Alternatively, you may have defined styles via JSS that compile into CSS and attached via the shadow stylesheet. To tell AMP which class to use, override the `shadowCss` property:
+
+```javascript
+import {CSS} from './my-element.jss';
+
+/** @override */
+AmpMyElement['shadowCss'] = CSS;
+```
+
+## Validation, Testing and Experiment
+
+### Allowing proper validation
+
+One of AMP's features is documentation validation checks to confirm it's valid AMP. When you implement your element, you must update the [AMP Validator](https://github.com/ampproject/amphtml/blob/main/validator/README.md)
+to include rules for your element. Otherwise documents using your extended component become invalid. Create your own rules by following the directions at
+[Contributing Component Validator Rules](https://github.com/ampproject/amphtml/blob/main/docs/component-validator-rules.md).
+
+When converting an existing component to Bento with a version bump, it suffices to copy existing validator tests if applicable for APIs that stay stable between versions. When copying test files, be sure to change the version of the extension being imported.
+
+In simple terms:
+
+1. Modifying the spec in the `.protoascii` file, using satisfies and excludes to differentiate restrictiveness between `0.1` and `1.0` versions
+2. Copying the existing `validator-amp-my-element.html` and `validator-amp-my-element.out` files from the `0.1/test` directory to the `1.0/test` directory
+3. Manually updating the script imported from the copied files to `1.0`
+
+In some cases, the validator rules for a `0.1` counterpart may allow for less strict usage of the component, such as with `requires_usage: EXEMPTED` or `deprecated_allow_duplicates: true`. The `1.0` version should reintroduce these restrictions by differentiating the versions via `satisfies` and `excludes`. [The validation PR for `amp-social-share`](https://github.com/ampproject/amphtml/pull/33478) serves as a useful reference and example for how to execute this change.
+
+For example, Bento `amp-soundcloud` .protoascii should be like:
+
+```protoascii
```

Comment:
```suggestion
```diff
```

Reply:
I recommend making this code block a diff and adding `+` in the lines that are added (recommended in a suggestion below) to make it clear how the file is changing.